### PR TITLE
plugins:foreman_azure - Release 1.0.0 (RPM)

### PR DESCRIPTION
**Pull request by dLobatog**
Can't be merged yet, it needs the fog-azure dependency & testing to see if it works with anything pre 1.11

### CODE REVIEW

File: rubygem-foreman_azure/rubygem-foreman_azure.spec [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+Summary: Azure integration with Foreman (theforeman.org)
+Name:    %{?scl_prefix}rubygem-%{gem_name}
+Version: 1.0.1
+Release: 1%{?foremandist}%{?dist}
+Group:   Applications/System
+License: GPLv3
+URL:     https://github.com/theforeman/foreman_azure
+Source0: http://rubygems.org/gems/%{gem_name}-%{version}.gem
+
+Requires: foreman >= 1.9.0
```
Comment (domcleal) [inline rubygem-foreman_azure/rubygem-foreman_azure.spec]: 1.11.0?

File: rubygem-foreman_azure/rubygem-foreman_azure.spec [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+Requires: %{?scl_prefix_ruby}ruby(release)
+Requires: %{?scl_prefix_ruby}rubygems
+Requires: %{?scl_prefix}rubygem(deface) < 2.0
+Requires: %{?scl_prefix}rubygem(fog-azure) = 0.0.2
```
Comment (domcleal) [inline rubygem-foreman_azure/rubygem-foreman_azure.spec]: I'd suggest loosening this exact restriction in future releases of the foreman_azure gem if you can.

File: rubygem-foreman_azure/rubygem-foreman_azure.spec [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+BuildRequires: %{?scl_prefix_ruby}rubygems
+BuildRequires: %{?scl_prefix}rubygem(deface) < 2.0
+BuildRequires: %{?scl_prefix}rubygem(fog-azure) = 0.0.2
+BuildRequires: foreman-plugin >= 1.11.0
+
+BuildArch: noarch
+
+Provides: %{?scl_prefix}rubygem(%{gem_name}) = %{version}
+Provides: foreman-plugin-azure
+%{?scl:Obsoletes: ruby193-rubygem-%{gem_name}}
```
Comment (domcleal) [inline rubygem-foreman_azure/rubygem-foreman_azure.spec]: Can be removed, we didn't ship a ruby193 version.

File: rubygem-foreman_azure/rubygem-foreman_azure.spec [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+%description
+Azure integration with Foreman.
+
+
+%package doc
+Summary: Documentation for %{name}
+Group: Documentation
+Requires: %{?scl_prefix}%{pkg_name} = %{version}-%{release}
+%{?scl:Obsoletes: ruby193-rubygem-%{gem_name}-doc}
```
Comment (domcleal) [inline rubygem-foreman_azure/rubygem-foreman_azure.spec]: ditto

File: rubygem-foreman_azure/rubygem-foreman_azure.spec [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+BuildRequires: %{?scl_prefix_ruby}rubygems-devel
+BuildRequires: %{?scl_prefix_ruby}rubygems
+BuildRequires: %{?scl_prefix}rubygem(deface) < 2.0
+BuildRequires: %{?scl_prefix}rubygem(fog-azure) = 0.0.2
+BuildRequires: foreman-plugin >= 1.11.0
+
+BuildArch: noarch
+
+Provides: %{?scl_prefix}rubygem(%{gem_name}) = %{version}
+Provides: foreman-plugin-azure
```
Comment (domcleal) [inline rubygem-foreman_azure/rubygem-foreman_azure.spec]: also provide `foreman-azure` so it looks like built in Foreman CRs

**Comment by domcleal** [comment]
This PR also needs asset precompilation adding to the build phase and %files, as foreman_azure ships a JS file.  You might want to try the gem2rpm template for Foreman plugins as it'll hopefully set it all up for you!

Please give it an hour or so after I merged the dependencies before rebasing this, as Koji needs to build everything.

File: rubygem-foreman_azure/rubygem-foreman_azure.spec (hunk 1 above)
Comment (dLobatog) [inline rubygem-foreman_azure/rubygem-foreman_azure.spec]: Why? fog-core 1.27 is the dependency I'm really looking for here and it's available since 1.9

Comment (domcleal) [inline rubygem-foreman_azure/rubygem-foreman_azure.spec]: Oh, I compared it to your Debian PR which stated >= 1.11.0.  The engine.rb actually states >= 1.8.

File: rubygem-foreman_azure/rubygem-foreman_azure.spec (hunk 5 above)
Comment (domcleal) [inline rubygem-foreman_azure/rubygem-foreman_azure.spec]: Unfixed?

**Comment by dLobatog** [comment]
Rebased & submitted a new spec based off the new gem2rpm spec

File: rubygem-foreman_azure/rubygem-foreman_azure.spec [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+BuildRequires: foreman-assets
+BuildRequires: foreman-plugin >= 1.9.0
+BuildRequires: %{?scl_prefix}rubygem(fog-azure) = 0.0.2
+BuildRequires: %{?scl_prefix}rubygem(deface) < 2.0
+BuildRequires: %{?scl_prefix_ruby}ruby(release)
+BuildRequires: %{?scl_prefix_ruby}ruby
+BuildRequires: %{?scl_prefix_ruby}rubygems-devel
+BuildArch: noarch
+Provides: %{?scl_prefix}rubygem(%{gem_name}) = %{version}
+Provides: foreman-plugin-%{plugin_name}
```
Comment (domcleal) [inline rubygem-foreman_azure/rubygem-foreman_azure.spec]: also provide foreman-azure so it looks like built in Foreman CRs

**Comment by dLobatog** [comment]
Updated to provide `foreman-azure` in addition to `foreman-plugin-azure`

**Comment by domcleal** [comment]
Thanks @dLobatog, building now.

**Comment by domcleal** [comment]
Packages will be available for nightly to begin with, then if they work OK then we can retag the dependency builds to 1.11/1.12 and rebuild foreman_azure if you wish?

**Comment by dLobatog** [comment]
Yup - that'd be perfect. Not sure if you're still building 1.10 but I believe it'd still work

**Comment by domcleal** [comment]
Ah yes, we are still building for 1.10 but that's a lot of work because I'd need to rebuild all of the dependencies.  I'll leave it for 1.11 and above if that's OK - which should appear in the repos later today.

**Pull request merged**